docs/subinterpreters.md

```diff
@@ -1,11 +1,13 @@
-# Sub-Interpreters
+# RedisGears Design: Python Sub-Interpreters
+RedisGears ships with an embedded [Python Interpreter](runtime.md#python-interpreter). This makes it possible to run Python code using the [`RG.PYEXECUTE` command](commands.md#rgpyexecute). Because the interpreter is a singleton that's shared among all calls to `RG.PYEXECUTE`, there's the risk of different executions using the same identifiers (e.g. global variables, functions or class names).
 
-RedisGears provides a complete embedded Python interpreter, which makes it possible to run Python code inside Redis using the `RG.PYEXECUTE` command. Because the interpreter is a singleton that's shared among all calls to `RG.PYEXECUTE`, there's the risk of different executions using the same identifiers (e.g. for global variables, function or class names).
+One way to address this problem by restarting the interpreter before each execution. However, an interpreter restart is a costly operation, time-wise, which renders this  approach less-than-desirable. Also, two executions may be running simultaneously, either in two different threads or even in the same thread (while the execution of the Python code itself is always non-parallelized, the rest of the execution's lifecycle may be). When more than one execution is running, restarting the interpreter is no longer an option. To address the issue correctly RedisGears make use of [Python's Sub-Interpreters](https://docs.python.org/3/c-api/init.html#sub-interpreter-support).
```

Comment:
this problem *is* by restarting
costly operation, time-wise, which > time-costly operation that
double-space before approach
may be > may be parellel

Reply:
Fixed via c28f1f0